membership_delegated_partner/tests/test_membership_delegate.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2019 Onestein - Andrea Stirpe
+# Copyright 2023 Tecnativa - Carolina Fernandez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields
+from odoo.tests import Form, TransactionCase
+
+
+class TestMembershipDelegate(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner1 = cls.env["res.partner"].create({"name": "Mr. Odoo"})
+        cls.partner2 = cls.env["res.partner"].create({"name": "Mrs. Odoo"})
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test membership product",
+                "membership": True,
+                "membership_date_from": "2017-01-01",
+                "membership_date_to": "2017-12-31",
+            }
+        )
+        cls.account = cls.env["account.account"].create(
+            {
+                "name": "Test account",
+                "code": "TEST",
+                "account_type": "asset_receivable",
+                "reconcile": True,
+            }
+        )
+        cls.journal_sale = cls.env["account.journal"].create(
+            {"name": "Test Sales Journal", "code": "tSAL", "type": "sale"}
+        )
+
+    def test_01_delegate(self):
+        """Delegates membership to partner 2"""
+        invoice = self.env["account.move"].create(
+            {
+                "name": "Test Customer Invoice",
+                "move_type": "out_invoice",
+                "partner_id": self.partner1.id,  # Invoicing partner
+                "delegated_member_id": self.partner2.id,  # Delegate membership to
+            }
+        )
+        self.env["account.move.line"].create(
```

Comment:
Instead of this (and doing the change from `invoice_line_ids[0]` to `line_ids[0]`, you should use the one2many field `invoice_line_ids` and create everything in one shot on the previous `account.move`>`create` call. The same for the other part where things are done the same way.